state-chain/pallets/cf-elections/src/electoral_systems/tests/solana_vault_swap_accounts.rs

```diff
@@ -0,0 +1,437 @@
+					(NEW_ACCOUNT_2, ()),
+					(NEW_ACCOUNT_3, ()),
+				]),
+				confirm_closed_accounts: BTreeSet::from([CLOSED_ACCOUNT_1]),
+			}),
+		);
+
+	TestSetup::<MinimalVaultSwapAccounts>::default()
+		.build_with_initial_election()
+		.expect_consensus(
+			generate_votes_specific_case([0, 0, 80, 80]),
+			Some(SolanaVaultSwapsVote {
+				new_accounts: BTreeSet::from([(NEW_ACCOUNT_3, ())]),
+				confirm_closed_accounts: BTreeSet::from([]),
+			}),
+		);
+
+	TestSetup::<MinimalVaultSwapAccounts>::default()
+		.build_with_initial_election()
+		.expect_consensus(ConsensusVotes { votes: vec![] }, None);
+
+	TestSetup::<MinimalVaultSwapAccounts>::default()
+		.build_with_initial_election()
+		.expect_consensus(generate_vote_no_consensus(), None);
+}
+
+fn generate_vote_no_consensus() -> ConsensusVotes<MinimalVaultSwapAccounts> {
+	let vote_1 = SolanaVaultSwapsVote {
+		new_accounts: BTreeSet::from([(1, ()), (2, ())]),
+		confirm_closed_accounts: BTreeSet::new(),
+	};
+
+	let vote_2 = SolanaVaultSwapsVote {
+		new_accounts: BTreeSet::from([(3, ()), (4, ())]),
+		confirm_closed_accounts: BTreeSet::new(),
+	};
+
+	ConsensusVotes {
+		votes: (0..80)
+			.map(|_| ConsensusVote { vote: Some(((), vote_1.clone())), validator_id: () })
+			.chain(
+				(0..80)
+					.map(|_| ConsensusVote { vote: Some(((), vote_2.clone())), validator_id: () }),
+			)
+			.collect::<Vec<_>>(),
+	}
+}
+
+fn generate_votes_specific_case(
+	no_of_each_vote: [usize; 4],
+) -> ConsensusVotes<MinimalVaultSwapAccounts> {
+	let vote_1 = SolanaVaultSwapsVote {
+		new_accounts: BTreeSet::from([
+			(NEW_ACCOUNT_1, ()),
+			(NEW_ACCOUNT_2, ()),
+			(NEW_ACCOUNT_3, ()),
+		]),
+		confirm_closed_accounts: BTreeSet::from([CLOSED_ACCOUNT_1, CLOSED_ACCOUNT_2]),
+	};
+
+	let vote_2 = SolanaVaultSwapsVote {
+		new_accounts: BTreeSet::from([(NEW_ACCOUNT_1, ()), (NEW_ACCOUNT_2, ())]),
+		confirm_closed_accounts: BTreeSet::from([CLOSED_ACCOUNT_1]),
+	};
+
+	let vote_3 = SolanaVaultSwapsVote {
+		new_accounts: BTreeSet::from([(NEW_ACCOUNT_1, ()), (NEW_ACCOUNT_3, ())]),
+		confirm_closed_accounts: BTreeSet::from([CLOSED_ACCOUNT_1]),
+	};
+
+	let vote_4 = SolanaVaultSwapsVote {
+		new_accounts: BTreeSet::from([(NEW_ACCOUNT_2, ()), (NEW_ACCOUNT_3, ())]),
+		confirm_closed_accounts: BTreeSet::from([CLOSED_ACCOUNT_2]),
+	};
+
+	ConsensusVotes {
+		votes: (0..no_of_each_vote[0])
+			.map(|_| ConsensusVote { vote: Some(((), vote_1.clone())), validator_id: () })
+			.chain(
+				(0..no_of_each_vote[1])
+					.map(|_| ConsensusVote { vote: Some(((), vote_2.clone())), validator_id: () }),
+			)
+			.chain(
+				(0..no_of_each_vote[2])
+					.map(|_| ConsensusVote { vote: Some(((), vote_3.clone())), validator_id: () }),
+			)
+			.chain(
+				(0..no_of_each_vote[3])
+					.map(|_| ConsensusVote { vote: Some(((), vote_4.clone())), validator_id: () }),
+			)
+			.collect::<Vec<_>>(),
+	}
+}
+
+fn generate_votes_for_account_range(n: u64, m: u64) -> SolanaVaultSwapsVote<Account, SwapDetails> {
```

Comment:
the main part on the comment here is that we can use the `Range` type rather than provide two values like this (there are a couple available, I think you want the core::ops::Range)